server/resp/src/main/java/org/infinispan/server/resp/commands/generic/RENAME.java

```diff
@@ -0,0 +1,68 @@
+package org.infinispan.server.resp.commands.generic;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiConsumer;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.server.resp.ByteBufPool;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * RENAME Resp Command
+ *
+ * @link <a href="https://redis.io/commands/rename/">RENAME</a>
+ * @since 15.0
+ */
+public class RENAME extends RespCommand implements Resp3Command {
+
+   public RENAME() {
+      super(3, 1, 2, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      byte[] srcKey = arguments.get(0);
+      byte[] dstKey = arguments.get(1);
+      return rename(handler, srcKey, dstKey, ctx, Consumers.OK_BICONSUMER);
+   }
+
+   public static CompletionStage<RespRequestHandler> rename(Resp3Handler handler, byte[] srcKey, byte[] dstKey,
+         ChannelHandlerContext ctx, BiConsumer<?, ByteBufPool> consumer) {
+      MediaType vmt = handler.cache().getValueDataConversion().getStorageMediaType();
+      final AdvancedCache<byte[], Object> acm = handler.cache().<byte[],Object>withMediaType(MediaType.APPLICATION_OCTET_STREAM, vmt);
+      return acm.removeAsyncEntry(srcKey).thenCompose(e -> {
+         if (e == null) {
+            RespErrorUtil.noSuchKey(handler.allocator());
+            return handler.myStage();
+         } else {
+            var timeService = handler.respServer().getTimeService();
+            // Immortal entry, copy metadata
+            if (e.getLifespan()<=0) {
+               var newMeta = e.getMetadata().builder();
+               return handler.stageToReturn(acm.putAsyncEntry(dstKey, e.getValue(), newMeta.build()).thenApply(ignore -> 1L), ctx,
```

Comment:
@rigazilla if the put fails after the remove we might lose the data, so I guess this is something we need to consider since we are unable to rename keys in a single op

Reply:
I added a warning in the doc comments